src-java/swmanager-topology/swmanager-storm-topology/src/test/java/org/openkilda/wfm/topology/switchmanager/mappers/GroupEntryConverterTest.java

```diff
@@ -0,0 +1,117 @@
+import org.openkilda.rulemanager.action.SetFieldAction;
+import org.openkilda.rulemanager.group.Bucket;
+import org.openkilda.rulemanager.group.GroupType;
+import org.openkilda.rulemanager.group.WatchGroup;
+import org.openkilda.rulemanager.group.WatchPort;
+
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+public class GroupEntryConverterTest {
+    public static final int GROUP_ID_VALUE = 1;
+    public static final long SWITCH_ID_VALUE = 2L;
+    public static final String MAC_ADDRESS_VALUE = "00:00:00:00:00:01";
+    public static final String IPV4_ADDRESS_VALUE = "255.255.255.255";
+    public static final int PORT_NUMBER_VALUE = 3;
+    public static final int SET_FIELD_VALUE = 4;
+    public static final int VNI_OVS_ACTION_VALUE = 7;
+    public static final int UDP_OVS_ACTION_VALUE = 8;
+
+    public static GroupId GROUP_ID = new GroupId(GROUP_ID_VALUE);
+    public static GroupType GROUP_TYPE = GroupType.ALL;
+    public static UUID RANDOM_UUID = UUID.randomUUID();
+    public static SwitchId SWITCH_ID = new SwitchId(SWITCH_ID_VALUE);
+    public static OfVersion OF_VERSION = OfVersion.OF_15;
+    public static MacAddress MAC_ADDRESS = new MacAddress(MAC_ADDRESS_VALUE);
+    public static IPv4Address IPV4_ADDRESS = new IPv4Address(IPV4_ADDRESS_VALUE);
+    public static ProtoConstants.PortNumber PORT_NUMBER = new ProtoConstants.PortNumber(PORT_NUMBER_VALUE);
+
+    public static PortOutAction PORT_OUT_ACTION = new PortOutAction(PORT_NUMBER);
+    public static SetFieldAction SET_FIELD_ACTION = new SetFieldAction(SET_FIELD_VALUE, Field.VLAN_VID);
+
+    public static PushVxlanAction PUSH_VXLAN_OVS_ACTION = PushVxlanAction.builder()
+            .type(ActionType.PUSH_VXLAN_OVS)
+            .vni(VNI_OVS_ACTION_VALUE)
+            .srcMacAddress(MAC_ADDRESS)
+            .dstMacAddress(MAC_ADDRESS)
+            .srcIpv4Address(IPV4_ADDRESS)
+            .dstIpv4Address(IPV4_ADDRESS)
+            .udpSrc(UDP_OVS_ACTION_VALUE)
+            .build();
+
+    public static GroupSpeakerData data;
+    public static Set<Action> writeActions = new HashSet<>();
+    public static List<Bucket> buckets = new LinkedList<>();
+
+    private void initializeData() {
+        writeActions.add(SET_FIELD_ACTION);
+        writeActions.add(PORT_OUT_ACTION);
+        writeActions.add(PUSH_VXLAN_OVS_ACTION);
+
+        buckets.add(new Bucket(WatchGroup.ALL, WatchPort.ANY, writeActions));
+
+        data = GroupSpeakerData.builder()
+                .uuid(RANDOM_UUID)
+                .switchId(SWITCH_ID)
+                .dependsOn(Collections.emptyList())
+                .ofVersion(OF_VERSION)
+                .groupId(GROUP_ID)
+                .type(GROUP_TYPE)
+                .buckets(buckets)
+                .build();
+    }
+
+    @Test
+    public void mapGroupEntryTest() {
+        initializeData();
```

Comment:
It's not good approach to initialize static test data by calling method in non static test. For example lets imagine what if we have 2 test cases? How you will initialize data for both of them?
1. If you will call `initializeData();` in each test case will initialize data twice and `writeActions` will contain 6 elements instead of 3.
2. if you will call `initializeData();` only in first test case then you wouldn't able to run test case 2 without test case 1. 

There are several good approaches for initialization. 
1. initialize elements immediately like you did with `PUSH_VXLAN_OVS_ACTION`
Example 
```public static Set<Action> writeActions = Sets.newHashSet(SET_FIELD_ACTION, PORT_OUT_ACTION, PUSH_VXLAN_OVS_ACTION);```
2. initialize immediately using static method:
```
    public static Set<Action> writeActions = initializeWriteActions();

    private static Set<Action> initializeWriteActions() {
        Set<Action> actions = new HashSet<>();
        actions.add(SET_FIELD_ACTION);
        actions.add(PORT_OUT_ACTION);
        actions.add(PUSH_VXLAN_OVS_ACTION);
        return actions;
    }
```
3. Initialize ones, using annotations

```   
 public static Set<Action> writeActions = new HashSet<>();

    @BeforeClass
    public static void initializeData() {
        writeActions.add(SET_FIELD_ACTION);
        writeActions.add(PORT_OUT_ACTION);
        writeActions.add(PUSH_VXLAN_OVS_ACTION);
        
        *****
```

annotation @BeforeClass will call this method ones, before running tests.
annotation @Before will call method before each test.

Please use one of this approaches in all other tests too

Reply:
Okay, thanks